cli/listener/src/main.rs

```diff
@@ -506,93 +506,109 @@ fn get_success_response(interval_prices: IntervalPrices) -> ResponseBox {
-        Err(_err) => {
-            return Some(get_error_response(ResponseError::new_bad_request(
-                "Error parsing Url",
-            )));
-        }
-    };
-    let method_name = parsed_url.path_segments()?.last()?;
-    if method_name != "apy" {
-        return Some(get_error_response(ResponseError::new_bad_request(
-            "Method not supported, use \"/apy?begin=<begin_date_iso8601>&end=<end_date_iso8601>\".",
-        )));
-    }
-    let parsed_request_url =
-        form_urlencoded::parse(parsed_url.query()?.as_bytes()).collect::<Vec<_>>();
-
-    let dates = match get_date_params(parsed_request_url) {
-        Ok(res) => Some(res),
-        Err(err) => return Some(get_error_response(err)),
-    }?;
-
+    range_date: &DateBeginEnd,
+) -> ResponseBox {
     let interval_prices = get_interval_price_for_period(
         db_connection
             .unchecked_transaction()
             .expect("Failed to create sqlite transaction."),
-        dates.start,
-        dates.end,
+        range_date.start,
+        range_date.end,
         SOLIDO_ID.to_owned(),
     );
     match interval_prices {
         // Error while getting the interval prices.
         Err(err) => {
             eprintln!("Internal Error when getting interval prices: {}", err);
-            Some(get_error_response(ResponseError::InternalServerError))
+            get_error_response(ResponseError::InternalServerError)
         }
         Ok(interval_prices_opt) => {
             if let Some(interval_prices) = interval_prices_opt {
                 // Got interval prices.
-                Some(get_success_response(interval_prices))
+                get_success_response(interval_prices)
             } else {
                 // No interval price could be calculated, probably because of few data points.
-                Some(get_error_response(ResponseError::new_bad_request(
+                get_error_response(ResponseError::new_bad_request(
                     "No data points for calculating the price interval.",
-                )))
+                ))
             }
         }
     }
 }
 
+enum ParseUrlResponse {
+    Metrics,
+    IntervalPriceRequest(DateBeginEnd), // Store date
+    Error(ResponseError),
+}
+
+fn parse_url(request_url: &str) -> Option<ParseUrlResponse> {
+    let parsed_url = match Url::parse(request_url) {
+        Ok(parsed_url) => parsed_url,
+        Err(_err) => return None,
+    };
+    let method_name = parsed_url.path_segments()?.last()?;
+    match method_name {
+        "apy" => {
+            let query_params =
+                form_urlencoded::parse(parsed_url.query()?.as_bytes()).collect::<Vec<_>>();
+            let dates = match get_date_params(query_params) {
+                Ok(res) => res,
+                Err(err) => return Some(ParseUrlResponse::Error(err)),
+            };
+            Some(ParseUrlResponse::IntervalPriceRequest(dates))
+        }
+        "metrics" => Some(ParseUrlResponse::Metrics),
```

Comment:
Now we will expose metrics on e.g. `/foo/bar/metrics`. If the reverse proxy does a blanked forward of `/api/apy*` -> `our-apy-daemon/api/apy/*`, it means that users can access the metrics on `/api/apy/metrics`. If we only accept `/metrics`, then we can keep them internal.